### PR TITLE
Add payroll periods

### PR DESCRIPTION
**Pull request by ddufresne**
https://isidor.numigi.net/web#id=16684&view_type=form&model=project.task&menu_id=200

### CODE REVIEW

File: timesheet_payroll_period/tests/test_timesheet.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import pytest
```
Comment (ddufresne) [inline timesheet_payroll_period/tests/test_timesheet.py]: Codacy Issue found: ['pytest' imported but unused](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/tests/test_timesheet.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+from ddt import ddt, data, unpack
+from datetime import datetime, timedelta
+from odoo.exceptions import AccessError
```
Comment (ddufresne) [inline timesheet_payroll_period/tests/test_timesheet.py]: Codacy Issue found: ['odoo.exceptions.AccessError' imported but unused](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/tests/test_timesheet.py (hunk 1 above)
Comment (ddufresne) [inline timesheet_payroll_period/tests/test_timesheet.py]: Codacy Issue found: [Unused import pytest](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/payroll_period.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from datetime import date
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline timesheet_payroll_period/models/payroll_period.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/account_analytic_line.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline timesheet_payroll_period/models/account_analytic_line.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/payroll_period.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+from odoo.exceptions import ValidationError
+
+
+def _recompute_timesheet_periods(env: 'Environment', date_from: date, date_to: date):
```
Comment (ddufresne) [inline timesheet_payroll_period/models/payroll_period.py]: Codacy Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/tests/test_timesheet.py (hunk 2 above)
Comment (ddufresne) [inline timesheet_payroll_period/tests/test_timesheet.py]: Codacy Issue found: [Unused AccessError imported from odoo.exceptions](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import (
```
Comment (ddufresne) [inline timesheet_payroll_period/models/__init__.py]: Codacy Issue found: ['.payroll_period' imported but unused](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/account_analytic_line.py (hunk 4 above)
Comment (ddufresne) [inline timesheet_payroll_period/models/account_analytic_line.py]: Codacy Issue found: ['odoo.exceptions.ValidationError' imported but unused](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/account_analytic_line.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline timesheet_payroll_period/models/account_analytic_line.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/payroll_period.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from datetime import date
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline timesheet_payroll_period/models/payroll_period.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/account_analytic_line.py (hunk 7 above)
Comment (ddufresne) [inline timesheet_payroll_period/models/account_analytic_line.py]: Codacy Issue found: ['odoo._' imported but unused](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: payroll_period/models/__init__.py [hunk 9]
```diff
@@ -0,0 +1,4 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import payroll_period
```
Comment (ddufresne) [inline payroll_period/models/__init__.py]: Codacy Issue found: ['.payroll_period' imported but unused](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

File: timesheet_payroll_period/models/payroll_period.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from datetime import date
+from odoo import api, fields, models
```
Comment (ddufresne) [inline timesheet_payroll_period/models/payroll_period.py]: Codacy Issue found: [Unused fields imported from odoo](https://app.codacy.com/app/numigi/odoo-timesheet-addons/pullRequest?prid=4692068)

**Review by foutoucour — APPROVED** [review]

**Comment by foutoucour** [comment]
👍

**Pull request merged**